### PR TITLE
bip: seed: Add an optional 'compressed_public' parameter to calculate_public_key_and_chaincode

### PR DESCRIPTION
**Pull request by xchapron-ledger**

### CODE REVIEW

File: src/ragger/bip/seed.py [hunk 1]
```diff
@@ -30,10 +30,10 @@ class CurveChoice(Enum):
...
 
 
-def calculate_public_key_and_chaincode(
-        curve: CurveChoice,
-        path: str,
-        mnemonic: Sequence[str] = SPECULOS_MNEMONIC) -> Tuple[str, str]:
+def calculate_public_key_and_chaincode(curve: CurveChoice,
+                                       path: str,
+                                       mnemonic: Sequence[str] = SPECULOS_MNEMONIC,
+                                       compressed_public: bool = False) -> Tuple[str, str]:
```
Comment (fbeutin-ledger) [inline src/ragger/bip/seed.py:36]: `compress_public_key` vu que c'est une demande

File: src/ragger/bip/seed.py [hunk 2]
```diff
@@ -43,4 +43,8 @@ def calculate_public_key_and_chaincode(
     public_key = child_node.PublicKey()
     chaincode = child_node.ChainCode()
 
-    return public_key.RawUncompressed().ToHex(), chaincode.ToHex()
+    if compressed_public:
+        public_key = public_key.RawCompressed()
```
Comment (fbeutin-ledger) [inline src/ragger/bip/seed.py:47]: `raw_public_key`

**Review by fbeutin-ledger — APPROVED** [review]
2 wording remarks but otherwise it looks good !

File: src/ragger/bip/seed.py (hunk 2 above)
Comment (xchapron-ledger) [reply src/ragger/bip/seed.py]: Done

File: src/ragger/bip/seed.py (hunk 1 above)
Comment (xchapron-ledger) [reply src/ragger/bip/seed.py]: done

**Review by fbeutin-ledger — APPROVED** [review]

**Pull request merged**